modules/core/src/gpu-type-utils/decode-texture-format.ts

```diff
@@ -20,16 +20,18 @@ export type DecodedTextureFormat = {
   components: 1 | 2 | 3 | 4;
   /** What is the data type of each component */
   dataType?: VertexType;
+  /** If this is a packed data type */
+  packed?: boolean;
   /** Number of bytes per pixel */
-  bpp?: number;
+  bytesPerPixel?: number;
+  /** Number of bits per channel (may be unreliable for packed formats) */
+  bitsPerChannel: number;
```

Comment:
Worth making this a `vec4`, so that supporting packed (or compressed!) formats in the future would be possible? Even if it is not accurate yet...

Reply:
Ha, yes that is a good idea. I'd probably add a separate field for that though as the most common use case is a common value and a vector gets fiddly for the app to work with, so will land this as is for now.